### PR TITLE
Upgrade source for navigation being used by foreign key attribute

### LINKED ISSUE #3832: ForeignKeyAttribute should upgrade sources for navigation based on which it was found
If explicit configuration is used to exclude a navigation in a relationship which is being pointed by `ForeignKey` attribute, we create relationship using that navigation by convention, while setting foreign key property on that which is found by convention, we match it with explicit relationship and the navigation is removed again. And this goes into infinite loop causing stack overflow.
Repro
````C#
public class Blog
    {
        public int Id { get; set; }
        
        public virtual IList<BlogMetadata> BlogMetadata { get; set; }
    }

    public class BlogMetadata
    {
        [ForeignKey("Blog")]
        public int Id { get; set; }
        
        public virtual Blog Blog { get; set; }
    }
protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            modelBuilder.Entity<BlogMetadata>().HasOne<Blog>().WithMany(b => b.BlogMetadata);
        }
````

### PR DESCRIPTION
**Pull request by smitpatel**
Resolves #3832

### CODE REVIEW

File: src/EntityFramework.Core/Metadata/Internal/InternalRelationshipBuilder.cs [hunk 1]
```diff
@@ -1212,11 +1218,25 @@ private bool CanSetPrincipalKey(IReadOnlyList<Property> properties, Configuratio
...
+                {
+                    return null;
+                }
+                var conflictingForeignKeyWithSameProperties = conflictingRelationships.FirstOrDefault(r => r.Metadata.Properties.SequenceEqual(dependentProperties));
+                if (conflictingForeignKeyWithSameProperties == null
+                    || !conflictingForeignKeyWithSameProperties.CanSetForeignKey(null, conflictingForeignKeyWithSameProperties.Metadata.DeclaringEntityType, configurationSource))
+                {
+                    return null;
+                }
+
```
Comment (AndriySvyryd) [inline src/EntityFramework.Core/Metadata/Internal/InternalRelationshipBuilder.cs]: We should also return null if there's any other conflicting foreign key that can't be removed

Comment (smitpatel) [inline src/EntityFramework.Core/Metadata/Internal/InternalRelationshipBuilder.cs]: Wouldn't above 2 ifs will take care of it?

Comment (smitpatel) [inline src/EntityFramework.Core/Metadata/Internal/InternalRelationshipBuilder.cs]: No, it wouldn't. :cry:

File: src/EntityFramework.Core/Metadata/Internal/InternalRelationshipBuilder.cs [hunk 2]
```diff
@@ -1540,11 +1560,24 @@ private bool CanSetPrincipalKey(IReadOnlyList<Property> properties, Configuratio
                     ModelBuilder.Metadata.ConventionDispatcher.OnForeignKeyRemoved(removedForeignKey.Item1, removedForeignKey.Item2);
                 }
 
+                if (newRelationshipBuilder != null
```
Comment (AndriySvyryd) [inline src/EntityFramework.Core/Metadata/Internal/InternalRelationshipBuilder.cs]: newRelationshipBuilder should never be null here

**Comment by AndriySvyryd** [comment]
:shipit:

**Pull request merged**